### ISSUE TITLE
How to send a blank message in chat

### ISSUE THREAD

**Issue by TinyMandalorian** [fuji-fabric/fuji#203]
Trying to make it so when you join the server it will clear the previous chat because every time i join it shows me previous messages from when i last logged in
![image](https://github.com/user-attachments/assets/aaf17ba4-4ed2-4a87-ae70-f4b46348325e)

so im trying to make it so when you join it will clear the chat so im sending blank messages but im only able to do that with the tellraw command because the send-message command doesnt allow me to send blank messages however when i try to use tellraw in the config it doesnt work because it requires quotation marks ive tried doubling the quotation marks around the text but it still doesnt work is there another way or a way to just stop this?

**Comment by sakurawald** [comment]
Disable the `chat.history` module.

**Comment by TinyMandalorian** [comment]
> Disable the `chat.history` module.

thanks that worked another question is there a way to make it so it doesnt send the welcome message twice when a player first joins it sends both the first join message and the join message im trying to get it to just send the first join message

**Comment by sakurawald** [comment]
> > Disable the `chat.history` module.
> 
> thanks that worked another question is there a way to make it so it doesnt send the welcome message twice when a player first joins it sends both the first join message and the join message im trying to get it to just send the first join message

That's expected behaviour.

- The event `on_player_first_joined` is only triggered when a player the first-time joins the server. (That's to say, its `left server times` = 0). This event is used to execute commands like `/kit give` and `/send-broadcast welcome message to this new player` .
- It's a bad idea to use `/send-broadcast` in `on_player_joined` event. If you want to modify the default joined message, just use `system_message module`, which allows you to custom the `join/leave message`.

In short:
1. If the joined player is `the first time join the server`, then the event `on_player_joined` and `on_player_first_joined` will be triggered.
2. If the joined player is **not** `the first time join the server`, then the event `on_player_joined` will be triggered.

**Comment by TinyMandalorian** [comment]
> > > Disable the `chat.history` module.
> > 
> > 
> > thanks that worked another question is there a way to make it so it doesnt send the welcome message twice when a player first joins it sends both the first join message and the join message im trying to get it to just send the first join message
> 
> That's expected behaviour.
> 
> * The event `on_player_first_joined` is only triggered when a player the first-time joins the server. (That's to say, its `left server times` = 0). This event is used to execute commands like `/kit give` and `/send-broadcast welcome message to this new player` .
> * It's a bad idea to use `/send-broadcast` in `on_player_joined` event. If you want to modify the default joined message, just use `system_message module`, which allows you to custom the `join/leave message`.
> 
> In short:
> 
> 1. If the joined player is `the first time join the server`, then the event `on_player_joined` and `on_player_first_joined` will be triggered.
> 2. If the joined player is **not** `the first time join the server`, then the event `on_player_joined` will be triggered.

Does system messages override styledchat join messages because I wanted to try and use the styledchat join messages and see if it did the same thing but i cant seem to get  it to enable

**Comment by sakurawald** [comment]
> > > > Disable the `chat.history` module.
> > > 
> > > 
> > > thanks that worked another question is there a way to make it so it doesnt send the welcome message twice when a player first joins it sends both the first join message and the join message im trying to get it to just send the first join message
> > 
> > 
> > That's expected behaviour.
> > 
> > * The event `on_player_first_joined` is only triggered when a player the first-time joins the server. (That's to say, its `left server times` = 0). This event is used to execute commands like `/kit give` and `/send-broadcast welcome message to this new player` .
> > * It's a bad idea to use `/send-broadcast` in `on_player_joined` event. If you want to modify the default joined message, just use `system_message module`, which allows you to custom the `join/leave message`.
> > 
> > In short:
> > 
> > 1. If the joined player is `the first time join the server`, then the event `on_player_joined` and `on_player_first_joined` will be triggered.
> > 2. If the joined player is **not** `the first time join the server`, then the event `on_player_joined` will be triggered.
> 
> Does system messages override styledchat join messages because I wanted to try and use the styledchat join messages and see if it did the same thing but i cant seem to get it to enable

Since the `styled chat` mod modifies the `player join message` and `player left message` by default, if you are using this mod, then you should not use `system_message module` to modify the `multiplayer.player.joined` and `multiplayer.player.left`.

What i don't understand is that, if you are using the `styled chat` mod, why not just modify the `join message` and `leave message` using it?

A related issue is: https://github.com/sakurawald/fuji/issues/65

**Comment by TinyMandalorian** [comment]
> > > > > Disable the `chat.history` module.
> > > > 
> > > > 
> > > > thanks that worked another question is there a way to make it so it doesnt send the welcome message twice when a player first joins it sends both the first join message and the join message im trying to get it to just send the first join message
> > > 
> > > 
> > > That's expected behaviour.
> > > 
> > > * The event `on_player_first_joined` is only triggered when a player the first-time joins the server. (That's to say, its `left server times` = 0). This event is used to execute commands like `/kit give` and `/send-broadcast welcome message to this new player` .
> > > * It's a bad idea to use `/send-broadcast` in `on_player_joined` event. If you want to modify the default joined message, just use `system_message module`, which allows you to custom the `join/leave message`.
> > > 
> > > In short:
> > > 
> > > 1. If the joined player is `the first time join the server`, then the event `on_player_joined` and `on_player_first_joined` will be triggered.
> > > 2. If the joined player is **not** `the first time join the server`, then the event `on_player_joined` will be triggered.
> > 
> > 
> > Does system messages override styledchat join messages because I wanted to try and use the styledchat join messages and see if it did the same thing but i cant seem to get it to enable
> 
> Since the `styled chat` mod modifies the `player join message` and `player left message` by default, if you are using this mod, then you should not use `system_message module` to modify the `multiplayer.player.joined` and `multiplayer.player.left`.
> 
> What i don't understand is that, if you are using the `styled chat` mod, why not just modify the `join message` and `leave message` using it?
> 
> A related issue is: #65

I just didnt know they had the messages till i opened it up and looked in detail I was at first using the command-event module but didnt like the way it worked for my purposes also i think styledchat just might have a bug with the join messages i disabled the system module and the join messages still dont appear

**Comment by TinyMandalorian** [comment]
![image](https://github.com/user-attachments/assets/d167c701-9234-4299-9810-e11c08298fbd)
well thats odd it shows it in console but just doesnt show it in chat

**Comment by sakurawald** [comment]
> > > > > > Disable the `chat.history` module.
> > > > > 
> > > > > 
> > > > > thanks that worked another question is there a way to make it so it doesnt send the welcome message twice when a player first joins it sends both the first join message and the join message im trying to get it to just send the first join message
> > > > 
> > > > 
> > > > That's expected behaviour.
> > > > 
> > > > * The event `on_player_first_joined` is only triggered when a player the first-time joins the server. (That's to say, its `left server times` = 0). This event is used to execute commands like `/kit give` and `/send-broadcast welcome message to this new player` .
> > > > * It's a bad idea to use `/send-broadcast` in `on_player_joined` event. If you want to modify the default joined message, just use `system_message module`, which allows you to custom the `join/leave message`.
> > > > 
> > > > In short:
> > > > 
> > > > 1. If the joined player is `the first time join the server`, then the event `on_player_joined` and `on_player_first_joined` will be triggered.
> > > > 2. If the joined player is **not** `the first time join the server`, then the event `on_player_joined` will be triggered.
> > > 
> > > 
> > > Does system messages override styledchat join messages because I wanted to try and use the styledchat join messages and see if it did the same thing but i cant seem to get it to enable
> > 
> > 
> > Since the `styled chat` mod modifies the `player join message` and `player left message` by default, if you are using this mod, then you should not use `system_message module` to modify the `multiplayer.player.joined` and `multiplayer.player.left`.
> > What i don't understand is that, if you are using the `styled chat` mod, why not just modify the `join message` and `leave message` using it?
> > A related issue is: #65
> 
> I just didnt know they had the messages till i opened it up and looked in detail I was at first using the command-event module but didnt like the way it worked for my purposes also i think styledchat just might have a bug with the join messages i disabled the system module and the join messages still dont appear

Can you describe the expected behaviour you want?
And also the configuration file for `system_message` (if you are using it), and `command event` (if you are using it).

**Comment by TinyMandalorian** [comment]
> > > > > > > Disable the `chat.history` module.
> > > > > > 
> > > > > > 
> > > > > > thanks that worked another question is there a way to make it so it doesnt send the welcome message twice when a player first joins it sends both the first join message and the join message im trying to get it to just send the first join message
> > > > > 
> > > > > 
> > > > > That's expected behaviour.
> > > > > 
> > > > > * The event `on_player_first_joined` is only triggered when a player the first-time joins the server. (That's to say, its `left server times` = 0). This event is used to execute commands like `/kit give` and `/send-broadcast welcome message to this new player` .
> > > > > * It's a bad idea to use `/send-broadcast` in `on_player_joined` event. If you want to modify the default joined message, just use `system_message module`, which allows you to custom the `join/leave message`.
> > > > > 
> > > > > In short:
> > > > > 
> > > > > 1. If the joined player is `the first time join the server`, then the event `on_player_joined` and `on_player_first_joined` will be triggered.
> > > > > 2. If the joined player is **not** `the first time join the server`, then the event `on_player_joined` will be triggered.
> > > > 
> > > > 
> > > > Does system messages override styledchat join messages because I wanted to try and use the styledchat join messages and see if it did the same thing but i cant seem to get it to enable
> > > 
> > > 
> > > Since the `styled chat` mod modifies the `player join message` and `player left message` by default, if you are using this mod, then you should not use `system_message module` to modify the `multiplayer.player.joined` and `multiplayer.player.left`.
> > > What i don't understand is that, if you are using the `styled chat` mod, why not just modify the `join message` and `leave message` using it?
> > > A related issue is: #65
> > 
> > 
> > I just didnt know they had the messages till i opened it up and looked in detail I was at first using the command-event module but didnt like the way it worked for my purposes also i think styledchat just might have a bug with the join messages i disabled the system module and the join messages still dont appear
> 
> Can you describe the expected behaviour you want? And also the configuration file for `system_message` (if you are using it), and `command event` (if you are using it). I assume that you are using styledchat now

I just want to make it so it doesnt send both join messages when a player first joins the server im trying to test it with styled chat now but styled chat join messages dont seem to be working and only sending the message in the console

**Comment by sakurawald** [comment]
Yeah, you can modify the message to what you want directly, in `styled chat`'s configuration.
```
      "joined_for_first_time": "<yellow>Hello ${player}</yellow>",
```

**Comment by TinyMandalorian** [comment]
> Yeah, you can modify the message to what you want directly, in `styled chat`'s configuration.
> 
> ```
>       "joined_for_first_time": "<yellow>Hello ${player}</yellow>",
> ```

Yep it just doesnt show up in the chat im wondering if that only sends the message before you can view chat and so you dont see it in game if I had a alt i could test this

**Comment by sakurawald** [comment]
> > > > > > > > Disable the `chat.history` module.
> > > > > > > 
> > > > > > > 
> > > > > > > thanks that worked another question is there a way to make it so it doesnt send the welcome message twice when a player first joins it sends both the first join message and the join message im trying to get it to just send the first join message
> > > > > > 
> > > > > > 
> > > > > > That's expected behaviour.
> > > > > > 
> > > > > > * The event `on_player_first_joined` is only triggered when a player the first-time joins the server. (That's to say, its `left server times` = 0). This event is used to execute commands like `/kit give` and `/send-broadcast welcome message to this new player` .
> > > > > > * It's a bad idea to use `/send-broadcast` in `on_player_joined` event. If you want to modify the default joined message, just use `system_message module`, which allows you to custom the `join/leave message`.
> > > > > > 
> > > > > > In short:
> > > > > > 
> > > > > > 1. If the joined player is `the first time join the server`, then the event `on_player_joined` and `on_player_first_joined` will be triggered.
> > > > > > 2. If the joined player is **not** `the first time join the server`, then the event `on_player_joined` will be triggered.
> > > > > 
> > > > > 
> > > > > Does system messages override styledchat join messages because I wanted to try and use the styledchat join messages and see if it did the same thing but i cant seem to get it to enable
> > > > 
> > > > 
> > > > Since the `styled chat` mod modifies the `player join message` and `player left message` by default, if you are using this mod, then you should not use `system_message module` to modify the `multiplayer.player.joined` and `multiplayer.player.left`.
> > > > What i don't understand is that, if you are using the `styled chat` mod, why not just modify the `join message` and `leave message` using it?
> > > > A related issue is: #65
> > > 
> > > 
> > > I just didnt know they had the messages till i opened it up and looked in detail I was at first using the command-event module but didnt like the way it worked for my purposes also i think styledchat just might have a bug with the join messages i disabled the system module and the join messages still dont appear
> > 
> > 
> > Can you describe the expected behaviour you want? And also the configuration file for `system_message` (if you are using it), and `command event` (if you are using it). I assume that you are using styledchat now
> 
> I just want to make it so it doesnt send both join messages when a player first joins the server im trying to test it with styled chat now but styled chat join messages dont seem to be working and only sending the message in the console

If you are using `styled chat`, then you should not using `system_message module` to hijack the `multiplayer.player.joined` message.

The reason is that: the `styled chat` mod will get and parse the text from `multiplayer.player.joined`, if you are using `system_message` module to hijack this message, then this message will be consumed **twice**. So when `styled chat` is ready to consume this text, it's not a translatable text, so here may be some problems.

**Comment by sakurawald** [comment]
> > Yeah, you can modify the message to what you want directly, in `styled chat`'s configuration.
> > ```
> >       "joined_for_first_time": "<yellow>Hello ${player}</yellow>",
> > ```
> 
> Yep it just doesnt show up in the chat im wondering if that only sends the message before you can view chat and so you dont see it in game if I had a alt i could test this

It's a defined behaviour in vanilla minecraft. You should start another minecraft client, i mean, the second account to join the server. And you will see the `joined message` in the first account.

**Comment by TinyMandalorian** [comment]
> > > Yeah, you can modify the message to what you want directly, in `styled chat`'s configuration.
> > > ```
> > >       "joined_for_first_time": "<yellow>Hello ${player}</yellow>",
> > > ```
> > 
> > 
> > Yep it just doesnt show up in the chat im wondering if that only sends the message before you can view chat and so you dont see it in game if I had a alt i could test this
> 
> It's a defined behaviour in vanilla minecraft. You should start another minecraft client, i mean, the second account to join the server. And you will see the `joined message` in the first account.

Ah ok thanks also I think it would be beneficial to open a discord server so we can ask weird questions like this instead of opening support tickets on github

**Comment by sakurawald** [comment]
> > > > Yeah, you can modify the message to what you want directly, in `styled chat`'s configuration.
> > > > ```
> > > >       "joined_for_first_time": "<yellow>Hello ${player}</yellow>",
> > > > ```
> > > 
> > > 
> > > Yep it just doesnt show up in the chat im wondering if that only sends the message before you can view chat and so you dont see it in game if I had a alt i could test this
> > 
> > 
> > It's a defined behaviour in vanilla minecraft. You should start another minecraft client, i mean, the second account to join the server. And you will see the `joined message` in the first account.
> 
> Ah ok thanks also I think it would be beneficial to open a discord server so we can ask weird questions like this instead of opening support tickets on github

The mod `styled chat` just works fine.

![image](https://github.com/user-attachments/assets/5973ae34-756f-4013-b87d-a015bea5f769)

**Comment by TinyMandalorian** [comment]
> > > > > Yeah, you can modify the message to what you want directly, in `styled chat`'s configuration.
> > > > > ```
> > > > >       "joined_for_first_time": "<yellow>Hello ${player}</yellow>",
> > > > > ```
> > > > 
> > > > 
> > > > Yep it just doesnt show up in the chat im wondering if that only sends the message before you can view chat and so you dont see it in game if I had a alt i could test this
> > > 
> > > 
> > > It's a defined behaviour in vanilla minecraft. You should start another minecraft client, i mean, the second account to join the server. And you will see the `joined message` in the first account.
> > 
> > 
> > Ah ok thanks also I think it would be beneficial to open a discord server so we can ask weird questions like this instead of opening support tickets on github
> 
> The mod `styled chat` just works fine.
> 
> ![image](https://private-user-images.githubusercontent.com/10484271/380491909-5973ae34-756f-4013-b87d-a015bea5f769.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMjk4NzAsIm5iZiI6MTczMDAyOTU3MCwicGF0aCI6Ii8xMDQ4NDI3MS8zODA0OTE5MDktNTk3M2FlMzQtNzU2Zi00MDEzLWI4N2QtYTAxNWJlYTVmNzY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDExNDYxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE1NmUwMjI0OWE2MTdjOGFkYTEwZjEzNGFhMzY2N2IwMTY4NmRiZDczMjQyMTZmMzY5OTE1YWU2NjQxNjNjYzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Frwhleks9FKKMrlFdMn8UYGDEBNd6ga7CCE3Miss-dQ)

is that what you saw when you logged in with a alt?

**Comment by sakurawald** [comment]
> > > > Yeah, you can modify the message to what you want directly, in `styled chat`'s configuration.
> > > > ```
> > > >       "joined_for_first_time": "<yellow>Hello ${player}</yellow>",
> > > > ```
> > > 
> > > 
> > > Yep it just doesnt show up in the chat im wondering if that only sends the message before you can view chat and so you dont see it in game if I had a alt i could test this
> > 
> > 
> > It's a defined behaviour in vanilla minecraft. You should start another minecraft client, i mean, the second account to join the server. And you will see the `joined message` in the first account.
> 
> Ah ok thanks also I think it would be beneficial to open a discord server so we can ask weird questions like this instead of opening support tickets on github



> > > > > > Yeah, you can modify the message to what you want directly, in `styled chat`'s configuration.
> > > > > > ```
> > > > > >       "joined_for_first_time": "<yellow>Hello ${player}</yellow>",
> > > > > > ```
> > > > > 
> > > > > 
> > > > > Yep it just doesnt show up in the chat im wondering if that only sends the message before you can view chat and so you dont see it in game if I had a alt i could test this
> > > > 
> > > > 
> > > > It's a defined behaviour in vanilla minecraft. You should start another minecraft client, i mean, the second account to join the server. And you will see the `joined message` in the first account.
> > > 
> > > 
> > > Ah ok thanks also I think it would be beneficial to open a discord server so we can ask weird questions like this instead of opening support tickets on github
> > 
> > 
> > The mod `styled chat` just works fine.
> > ![image](https://private-user-images.githubusercontent.com/10484271/380491909-5973ae34-756f-4013-b87d-a015bea5f769.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMjk4NzAsIm5iZiI6MTczMDAyOTU3MCwicGF0aCI6Ii8xMDQ4NDI3MS8zODA0OTE5MDktNTk3M2FlMzQtNzU2Zi00MDEzLWI4N2QtYTAxNWJlYTVmNzY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDExNDYxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE1NmUwMjI0OWE2MTdjOGFkYTEwZjEzNGFhMzY2N2IwMTY4NmRiZDczMjQyMTZmMzY5OTE1YWU2NjQxNjNjYzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Frwhleks9FKKMrlFdMn8UYGDEBNd6ga7CCE3Miss-dQ)
> 
> is that what you saw when you logged in with a alt?

No, a new joined player can't see the `multiplayer.player.joined` message.
However, he can see the messages sent by the `on_player_first_joined` event and `on_player_joined` event.

The reason is that: the point `multiplayer.player.joined` message sent is too early to let the new joined player to see.

**Comment by TinyMandalorian** [comment]
> > > > > Yeah, you can modify the message to what you want directly, in `styled chat`'s configuration.
> > > > > ```
> > > > >       "joined_for_first_time": "<yellow>Hello ${player}</yellow>",
> > > > > ```
> > > > 
> > > > 
> > > > Yep it just doesnt show up in the chat im wondering if that only sends the message before you can view chat and so you dont see it in game if I had a alt i could test this
> > > 
> > > 
> > > It's a defined behaviour in vanilla minecraft. You should start another minecraft client, i mean, the second account to join the server. And you will see the `joined message` in the first account.
> > 
> > 
> > Ah ok thanks also I think it would be beneficial to open a discord server so we can ask weird questions like this instead of opening support tickets on github
> 
> > > > > > > Yeah, you can modify the message to what you want directly, in `styled chat`'s configuration.
> > > > > > > ```
> > > > > > >       "joined_for_first_time": "<yellow>Hello ${player}</yellow>",
> > > > > > > ```
> > > > > > 
> > > > > > 
> > > > > > Yep it just doesnt show up in the chat im wondering if that only sends the message before you can view chat and so you dont see it in game if I had a alt i could test this
> > > > > 
> > > > > 
> > > > > It's a defined behaviour in vanilla minecraft. You should start another minecraft client, i mean, the second account to join the server. And you will see the `joined message` in the first account.
> > > > 
> > > > 
> > > > Ah ok thanks also I think it would be beneficial to open a discord server so we can ask weird questions like this instead of opening support tickets on github
> > > 
> > > 
> > > The mod `styled chat` just works fine.
> > > ![image](https://private-user-images.githubusercontent.com/10484271/380491909-5973ae34-756f-4013-b87d-a015bea5f769.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMjk4NzAsIm5iZiI6MTczMDAyOTU3MCwicGF0aCI6Ii8xMDQ4NDI3MS8zODA0OTE5MDktNTk3M2FlMzQtNzU2Zi00MDEzLWI4N2QtYTAxNWJlYTVmNzY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDExNDYxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE1NmUwMjI0OWE2MTdjOGFkYTEwZjEzNGFhMzY2N2IwMTY4NmRiZDczMjQyMTZmMzY5OTE1YWU2NjQxNjNjYzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Frwhleks9FKKMrlFdMn8UYGDEBNd6ga7CCE3Miss-dQ)
> > 
> > 
> > is that what you saw when you logged in with a alt?
> 
> No, a new joined player can't see the `multiplayer.player.joined` message. However, he can see the messages sent by the `on_player_first_joined` event and `on_player_joined` event.
> 
> The reason is that: the point `multiplayer.player.joined` message sent is too early to let the new joined player to see.

Ok that makes sense thank you

**Comment by sakurawald** [comment]
This screenshot is captured from an account the first-time joined the server.
![image](https://github.com/user-attachments/assets/0f5e3e9a-79b4-4721-8247-e78c9f1c1386)

As expected behavior defined by mojang, he can't see the `multiplayer.player.joined` message.

**Comment by sakurawald** [comment]
You should use the configuration in `styled chat` mod:
```
      "joined_the_game": "<gray>✚</gray> <color:#85ff8f><lang:multiplayer.player.joined:'${player}'>",
      "joined_for_first_time": "<yellow>Hello ${player}</yellow>",
```

The mod `styled chat` will handle it well. 

Don't hijack the message like `multiplayer.player.joined` using `system_message module` if this message is handled by `styled chat`.
